File: src/modules/python/test/test001_smoke.py
# mge - Modern Game Engine
# Copyright (c) 2021 by Alexander Schroeder
# All rights reserved.
import unittest
import mge


class TestSmoke(unittest.TestCase):

    def test_simple_construct(self):
        """Tests default construction"""
        v = mge.fvec2()

    def test_kwargs_unsupported(self):
        """Tests that keyword args are currently unsupported"""
        with self.assertRaises(SystemError):
            v = mge.fvec2(bubba=17)

    def test_default_field_access(self):
        """Test a default constructor"""
        v = mge.fvec2()
        self.assertEqual(0.0, v.x)
        self.assertEqual(0.0, v.y)

    def test_set_field_successful(self):
        """Test setting a field successfully"""
        p = mge.point()
        p.x = 17
        self.assertEqual(17, p.x)

    def test_set_field_unsuccessful(self):
        """Test setting a field successfully"""
        p = mge.point()
        with self.assertRaises(ValueError):
            p.x = 'hossa'

    def test_application_type_exists(self):
        """Test existence of application type"""
        x = mge.application
        self.assertEqual("<class 'application'>", repr(x))

    def test_application_subtype_exists(self):
        """Test existence of application type"""
        x = mge.application.input_listener
        self.assertEqual(
            "<class 'class std::function<void __cdecl(void)>'>", repr(x))

    def test_application_similar_subtype_exists(self):
        """Test existence of application type"""
        x = mge.application.input_listener
        y = mge.application.quit_listener
        assert x is not None and y is not None

    def test_method_call(self):
        """Test simple method call"""
        vf = mge.vertex_format(mge.data_type.FLOAT, 3)
        assert vf.binary_size() == 12


if __name__ == "main":
    unittest.main()

File: src/modules/lua/test/test001_smoke.lua
unittest = require "unittest"

function test_builtin_module()
    assert(type(package.loaded["__mge__"]) == "table", "mge module not loaded")
end

function test_mge_module_exists()
   local mod = package.loaded["mge"]
   assert(type(mod) == "table", "mge module seems not be loaded")
   assert(type(mge) == "table", "mge module not accessible by global var")
end

-- function test_mge_data_type_exists()
--     assert(type(mge.data_type) == "table")
--     assert(mge.data_type.HALF == 11)
-- end

-- function test_mge_point_type_exists()
--     local mge = package.loaded["mge"]
--     local point_type = mge.point
--     assert(type(point_type) == "table", "point type is not accessible")
-- end

-- function test_mge_point_construct()
--     local mge = package.loaded["mge"]
--     p = mge.point(123, 456)
--     assert(type(p) == "userdata", "point not created as wanted")
--     assert(p.x == 123, "x coordinate not set")
--     assert(p.y == 456, "y coordinate not set")
--     p = nil
--     collectgarbage()

-- end

-- function test_application_similar_subtype_exists()
--     local mge = package.loaded["mge"]
--     local il_type = mge.application.input_listener
--     local ql_type = mge.application.quit_listener
--     assert(type(il_type) == "table", "input_listener type is not accessible")
--     assert(type(ql_type) == "table", "quit_listener type is not accessible")
-- end

-- function test_method_call()
--     local vf = mge.vertex_format(mge.data_type.FLOAT, 3)
--     assert(type(vf) == "userdata", "cannot create object")
--     assert(vf:binary_size() == 12, "cannot call simple method")
-- end

unittest.main()